p2p/options.go

```diff
@@ -1,30 +1,55 @@
 package p2p
 
 import (
+	"fmt"
+
 	libp2p "github.com/libp2p/go-libp2p"
 	libp2pConfig "github.com/libp2p/go-libp2p/config"
+	ma "github.com/multiformats/go-multiaddr"
 )
 
+func defaultListenAddrs(p2pPort string) []ma.Multiaddr {
+	multiaddrs := []ma.Multiaddr{}
+	local1, _ := ma.NewMultiaddr(fmt.Sprintf("/ip4/0.0.0.0/tcp/%s", p2pPort))
+	local2, _ := ma.NewMultiaddr(fmt.Sprintf("/ip4/127.0.0.1/tcp/%s", p2pPort))
+	multiaddrs = append(multiaddrs, local1)
+	multiaddrs = append(multiaddrs, local2)
+	return multiaddrs
+}
+
+// CreateOptions queries p2p.Config for any user-submitted options and assigns defaults
 func (p2p *P2p) CreateOptions() []libp2pConfig.Option {
 	options := []libp2pConfig.Option{}
-	if p2p.Config.GetBool("p2p.options.enableDHT") {
-		options = append(options, p2p.initDHT())
-	}
+	externalIP := p2p.Config.GetString("p2p.externalIP")
+	p2pPort := p2p.Config.GetString("p2p.port")
 
-	if p2p.Config.GetBool("p2p.options.enableIdentity") {
-		options = append(options, libp2p.Identity(p2p.privateKey))
-	}
+	options = append(options, p2p.initDHT())
+	options = append(options, libp2p.Identity(p2p.privateKey))
 
-	if p2p.Config.GetBool("p2p.options.enableRelay") {
+	if p2p.Config.GetBool("p2p.enableRelay") {
 		options = append(options, libp2p.EnableRelay())
 	}
 
-	if p2p.Config.GetBool("p2p.options.enableAutoRelay") {
+	if p2p.Config.GetBool("p2p.enableAutoRelay") {
 		options = append(options, libp2p.EnableAutoRelay())
 	}
 
-	if p2p.Config.GetBool("p2p.options.enableNATPortMap") {
+	if p2p.Config.GetBool("p2p.enableNATPortMap") {
 		options = append(options, libp2p.NATPortMap())
+	} else {
+		multiaddrs := defaultListenAddrs(p2pPort)
```

Comment:
I would create functions for these.

Reply:
They are on the cusp of being somewhat ugly here, but I wouldn't separate them because they are so dependant on variables inside the method.